enterprise/server/test/integration/remote_execution/rbetest/rbetest.go

```diff
@@ -381,6 +381,71 @@ func (s *BuildBuddyServer) GRPCPort() int {
 	return s.port
 }
 
+func (s *BuildBuddyServer) GRPCAddress() string {
+	return fmt.Sprintf("grpc://localhost:%d", s.GRPCPort())
+}
+
+const (
+	testCommandStateUnknown = iota
+	testCommandStateStarted
+	testCommandStateDisconnected
+)
+
+type testCommandState struct {
+	mu           sync.Mutex
+	opChan       chan *retpb.ControllerToCommandRequest
+	state        int
+	stateChanged chan struct{}
+}
+
+func (s *testCommandState) SetState(state int) {
+	s.mu.Lock()
+	s.state = state
+	if s.stateChanged != nil {
+		s.stateChanged <- struct{}{}
```

Comment:
would it make sense to remove teh mutex and just shove the state down the pipe to your waiter (which can update the member variable if you need that?)

Reply:
Making the channel optional means we don't need to worry about whether or not there is currently a waiter on the channel. We could make the channel buffered, but then we need to worry about setting the capacity to make sure the channel doesn't get filled. Don't think we should get too caught up on this given this is test code.